src/SensioLabs/Melody/Console/Command/RunCommand.php

```diff
@@ -113,6 +118,50 @@ protected function execute(InputInterface $input, OutputInterface $output)
 
                 $userConfig->addTrustedSignature($e->getResource()->getSignature());
                 $configRepository->save($userConfig);
+            } catch (AuthenticationRequiredException $e) {
+                $output->writeln(sprintf(
+                    '<error>The following error was encountered while trying to access the resource: "%s"</error>',
+                    $e->getMessage()
+                ));
+
+                $handler = $e->getHandler();
+                if (!$handler instanceof AuthenticableHandlerInterface) {
+                    $output->writeln(sprintf('<error>%s.</error>', $e->getMessage()));
+
+                    return 1;
+                }
+                $this->askCredentials($handler, $userConfig, $input, $output);
+            }
+        }
+    }
+
+    private function askCredentials(AuthenticableHandlerInterface $handler, UserConfiguration $userConfig, InputInterface $input, OutputInterface $output)
+    {
+        // Normalize required credentials format:
+        $requiredCredentials = $handler->getRequiredCredentials();
+        if (ctype_digit(implode('', array_keys($requiredCredentials)))) {
```

Comment:
So here, you are trying to determine if `$requiredCredentials` is something like:
```
['username', 'password']
```
?